op-node/chaincfg/chains.go

```diff
@@ -2,135 +2,100 @@ package chaincfg
-	BatchInboxAddress:      common.HexToAddress("0xff00000000000000000000000000000011155420"),
-	DepositContractAddress: common.HexToAddress("0x16fc5058f25648194471939df75cf27a2fdc48bc"),
-	L1SystemConfigAddress:  common.HexToAddress("0x034edd2a225f7f429a63e0f1d2084b9e0a93b538"),
-	RegolithTime:           u64Ptr(0),
-}
+var Mainnet, Goerli, Sepolia *rollup.Config
 
-var NetworksByName = map[string]rollup.Config{
-	"goerli":  Goerli,
-	"mainnet": Mainnet,
-	"sepolia": Sepolia,
+func init() {
+	mustCfg := func(name string) *rollup.Config {
+		cfg, err := GetRollupConfig(name)
+		if err != nil {
+			panic(fmt.Errorf("failed to load rollup config %q: %w", name, err))
+		}
+		return cfg
+	}
+	Mainnet = mustCfg("op-mainnet")
+	Goerli = mustCfg("op-goerli")
+	Sepolia = mustCfg("op-sepolia")
 }
 
-var L2ChainIDToNetworkName = func() map[string]string {
+var L2ChainIDToNetworkDisplayName = func() map[string]string {
 	out := make(map[string]string)
-	for name, netCfg := range NetworksByName {
-		out[netCfg.L2ChainID.String()] = name
+	for _, netCfg := range superchain.OPChains {
+		out[fmt.Sprintf("%d", netCfg.ChainID)] = netCfg.Name
 	}
 	return out
 }()
 
+// AvailableNetworks returns the selection of network configurations that is available by default.
+// Other configurations that are part of the superchain-registry can be used with the --beta.network flag.
 func AvailableNetworks() []string {
+	return []string{"op-mainnet", "op-goerli", "op-sepolia"}
+}
+
+// BetaAvailableNetworks returns all available network configurations in the superchain-registry.
+// This set of configurations is experimental, and may change at any time.
+func BetaAvailableNetworks() []string {
 	var networks []string
-	for name := range NetworksByName {
-		networks = append(networks, name)
+	for _, cfg := range superchain.OPChains {
+		networks = append(networks, cfg.Chain+"-"+cfg.Superchain)
 	}
 	return networks
 }
 
-func GetRollupConfig(name string) (rollup.Config, error) {
-	network, ok := NetworksByName[name]
-	if !ok {
-		return rollup.Config{}, fmt.Errorf("invalid network %s", name)
+func IsAvailableNetwork(name string, beta bool) bool {
+	name = handleLegacyName(name)
+	available := AvailableNetworks()
+	if beta {
+		available = BetaAvailableNetworks()
 	}
+	for _, v := range available {
+		if v == name {
+			return true
+		}
+	}
+	return false
+}
+
+func handleLegacyName(name string) string {
+	switch name {
+	case "goerli":
+		return "op-goerli"
+	case "mainnet":
+		return "op-mainnet"
+	case "sepolia":
+		return "op-sepolia"
+	default:
+		return name
+	}
+}
 
-	return network, nil
+// ChainByName returns a chain, from known available configurations, by name.
+// ChainByName returns nil when the chain name is unknown.
+func ChainByName(name string) *superchain.ChainConfig {
+	// Handle legacy name aliases
+	name = handleLegacyName(name)
+	for _, chainCfg := range superchain.OPChains {
+		if strings.EqualFold(chainCfg.Chain+"-"+chainCfg.Superchain, name) {
```

Comment:
What is `chainCfg.Chain` & `chainCfg.Superchain`?

Reply:
`Chain` is the simple lowercase identifier of the chain (without superchain part, e.g. `op`, `base`, `zora`, etc.) and `chainCfg.Superchain` is the superchain identifier (`mainnet`, `sepolia`, `goerli`, etc.). And there's `chainCfg.Name` for the prettified display name (e.g. `OP-Mainnet`).

Reply:
One callout here is that I wasn't able to figure this out by reading the chainconfig comments. https://github.com/ethereum-optimism/superchain-registry/blob/5d3ca1966435827ac8c4216bddcb541ec64c1d6c/superchain/superchain.go#L33-L51